encoder.py

```diff
@@ -136,12 +139,17 @@ def _get_docs_batch_generator(self, docs: DocumentArray, parameters: Dict):
     def _compute_embeddings(self, docs_batch_generator: Iterable) -> None:
         with torch.no_grad():
             for document_batch in docs_batch_generator:
-                blob_batch = np.stack([d.blob for d in document_batch])
-                images = self._maybe_move_channel_axis(blob_batch)
+                blob_batch = [d.blob for d in document_batch]
+                if self.use_default_preprocessing:
```

Comment:
I am not so sure these transformations can work in batches.

Reply:
They are applied per image
```python
def _preprocess_image(self, images: List[np.array]) -> List[np.ndarray]:
    return [self._preprocess(img) for img in images]
```